filebeat/module/system/syslog/ingest/pipeline.json

```diff
@@ -42,8 +43,14 @@
         {
             "date": {
                 "if": "ctx.event.timezone != null",
-                "field": "@timestamp",
-                "formats": ["ISO8601"],
+                "field": "system.syslog.timestamp",
+                "target_field": "@timestamp",
+                "formats": [
+                    "MMM  d HH:mm:ss",
+                    "MMM dd HH:mm:ss",
+                    "MMM d HH:mm:ss",
+                    "ISO8601"
```

Comment:
I think that timestamp will be incorrect for logs parsed as `ISO8601` if we also apply the `"timezone": "{{ event.timezone }}"`.

One thing we could do is to refactor date parsing here to explictly check if the data is in ISO8601 format or not, doing something like this (not tested):
```
"date": {
    "field": "system.syslog.timestamp",
    "target_field": "_tmp_timestamp",
    "formats": ["ISO8601"],
    "ignore_failure": true
},
"date": {
    "if": "ctx._tmp_timestamp == nil && event.timezone != nil",
    "field": "system.syslog.timestamp",
    "target_field": "_tmp_timestamp",
    "timezone": "{{ event.timezone }}",
    "formats": [
        "MMM  d HH:mm:ss",
        "MMM dd HH:mm:ss",
        "MMM d HH:mm:ss"
    ]
},
"date": {
    "if": "ctx._tmp_timestamp == nil",
    "field": "system.syslog.timestamp",
    "target_field": "_tmp_timestamp",
    "formats": [
        "MMM  d HH:mm:ss",
        "MMM dd HH:mm:ss",
        "MMM d HH:mm:ss"
    ]
},
"remove": {
    "fields": ["@timestamp"]
},
"rename". {
    "field": "_tmp_timestamp",
    "target_field": "@timestamp"
},
```
What do you think?

Reply:
Hmm, I think the `@timestamp` field will still be correct even if `system.syslog.timestamp` is already in ISO8601.
For example, if the system.syslog.timestamp is `2019-08-26T03:00:00Z`, and event.timezone is `+08:00`, then the result of this date processor will be: `2019-07-26T11:00:00+0800`, which is still right, just in another timezone as specified by event.timezone.

Or, do we want to convert all these @timestamp back  to UTC timezone? which means add one more date processor with timezone set to `UTC` .